internals/plan/plan.go

```diff
@@ -553,9 +523,7 @@ func (t *LogTarget) Merge(other *LogTarget) {
 	if other.Location != "" {
 		t.Location = other.Location
 	}
-	if other.Selection != "" {
-		t.Selection = other.Selection
-	}
+	t.Services = append(t.Services, other.Services...)
```

Comment:
I'm not sure if we should "reduce" the services list when merging. For example,
```
[svc1, svc2]  <--  [-svc1]       =  [svc1, svc2, -svc1]       ==>  [svc2]
[svc1, svc2]  <--  [all, -svc3]  =  [svc1, svc2, all, -svc3]  ==>  [all, -svc3]
[svc1, svc2]  <--  [-all, svc3]  =  [svc1, svc2, -all, svc3]  ==>  [svc3]
```

The way I've implemented `LogsTo`, it's not *necessary* to do this, but it might help to keep things more manageable (e.g. if we have to remarshal due to the user running `pebble plan`). It will complicate the logic though.

Reply:
I think what you've done here is the right approach. The field itself behaves according to the standard merge logic, so `pebble plan` should show the full list of merged items (rather than the evaluated result).

Separately we could consider adding a `pebble log-targets` or similar command which would show the status of the log targets: what services they've been evaluated as actually going to, how many logs sent, number of errors, that sort of thing. Future work...

Reply:
> The field itself behaves according to the standard merge logic, so `pebble plan` should show the full list of merged items

Yes, but that's a **choice** we've made here. I'm wondering if it's the right choice.

It would be equally valid to say "when merging, remove duplicates / reduce the list to a minimal equivalent set". (We previously did this with the `services.log-targets` field). It would behave the same either way, so I guess it's a matter of taste.

Reply:
I think leaving the merge without evaluating it also help with debugging. If someone does not understand why a particular service disappeared, the complete history is there. If you evaluate it, its gone.

Reply:
I am happy keeping this simple, we don't need to prematurely optimise here.